core/src/main/java/org/apache/carbondata/core/datamap/dev/expr/DataMapExprWrapper.java

```diff
@@ -32,14 +32,14 @@
  * It is the wrapper around datamap and related filter expression. By using it user can apply
  * datamaps in expression style.
  */
-public interface DataMapExprWrapper extends Serializable {
+public abstract class DataMapExprWrapper implements Serializable {
```

Comment:
you can still use interface, java 8 interface can have default implementation

Reply:
since some users are still using older java version,i think we can keep until we completely move out.